templates/default/mysql-job.sh.erb

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# rdiff-backup MySQL job script for $JOB managed by Chef. Changes will be overwritten.
+
+# This script is normally run by cron, but can be run manually to start the job if it is not already running.
+
+FILE=`basename $0`
+DIR=`dirname $0`
+PIDFILE="$DIR/.$FILE.lock"
+JOB="${FILE%.sh}"
+LOGPATH='/var/log/rdiff-backup'
+INSTANCES=`pgrep -fu <%= @suser %> $FILE | wc -l`
+
+# Prepends date to every line of stdin, then writes to both stdout and the log file.
+function log() {
+  sed "s/^/[`date +'%Y-%m-%d %H:%M:%S'`] /" | tee >(/usr/sbin/cronolog "${LOGPATH}/${JOB}/%Y-%m.log")
+}
+
+# Prepends date to every line of stdin, then writes to stdout and both the individual and general log files.
+function logboth() {
+  sed "s/^/[`date +'%Y-%m-%d %H:%M:%S'`] /" | tee >(/usr/sbin/cronolog "${LOGPATH}/${JOB}/%Y-%m.log") >(/usr/sbin/cronolog "${LOGPATH}/general/%Y-%m-%d.log" -S "${LOGPATH}/current.log")
+}
+
+# If we're root instead of the rdiff-backup server user, rerun the script as it.
+if [ `whoami` = 'root' ]; then
+  su <%= @suser %> -c $0
+else
+
+  # Only run if this script isn't already running
+  if [ $INSTANCES -lt 3 ]; then
+    echo $$ > $PIDFILE 2> >(log)
```

Comment:
No, I wasn't intending to print the pid to the log, only any errors that might have occurred.  Though I don't know if that's doing anything either since I don't know that echo or > can print errors.